### DISCUSSION TITLE
How to pass variables to resolveComponent for Dynamic Components from props

### DISCUSSION THREAD

**Discussion by aligzl** [nuxt/nuxt#16843] [Questions]
Hello,

How can i load dynamic components which variable will pass from props  or any .

This is what i tested.

```
<template>
  <li class="mt-3 p-2 target:rounded-lg hover:text-blue-300">
    <nuxt-link :to="to" exact exact-active-class="text-blue-300" class="flex flex-col items-center">
      <component :is="svgName" />
      <span class="mt-2">{{ title }}</span>
    </nuxt-link>
  </li>
</template>
<script setup>
  const props = defineProps({
    title: {
      type: String,
      readonly: true,
      default: ""
    },
    to: {
      type: String,
      default: "/admin",
    },
    svg: {
      type: String,
      default: "svg-home",
    }
  })

  const svgName = resolveComponent(props.svg)

</script>
```

i got fallowing error:

`Failed to resolve component: svg-home
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. `

Thanks

**Comment by wobsoriano** [comment]
Does it really have to be wrapped in a `resolveComponent` function or have you tried using the prop directly like

```ts
<component :is="svg" />
```

**Reply by aligzl** [reply]
> Does it really have to be wrapped in a `resolveComponent` function or have you tried using the prop directly like
> 
> ```ts
> <component :is="svg" />
> ```

nope it doesnt work like that

**Reply by wobsoriano** [reply]
How about a computed value then? And are you sure have that icon in the same directory?

**Reply by aligzl** [reply]
ok the problem was belongs to vite limitations.

**Comment by ilyaDegtyarenko** [comment]
Same error **[Vue warn]: Failed to resolve component: AppIconChat**

_**script setup**_
```
type Props = {
    is: string
}

const props = defineProps<Props>(),
    component = resolveComponent(props.is) // AppIconChat
```

_**template**_
```
<div>
    <component :is="component"/> - not works
    <AppIconChat/> - works
</div>
```

**Reply by mydoum** [reply]
I assume this is for Nuxt 3
It must be a [global component in order to work](https://v3.nuxtjs.org/guide/directory-structure/components/#dynamic-components)
The rules are : 

- resolveComponent('FooBar') Would work with a component at` ./components/foo/Bar.vue`, and it must be a string, not with backticks (``), nor with a variable
-  resolveComponent(myVariable) Would work either if the component is global

For this last rule, you can define a global component with two ways.
The first way, is to put all your components globals (not recommended)
```
export default defineNuxtConfig({
    components: {
     global: true,
     dirs: ['~/components']
    },
  })
```
OR you puts your targeted component inside the `./components/global`
For example you add a component at `./component/global/Bar.vue`
You can retreive it with
```
<template>
  <component :is="dynamicComponent" />
</template>
<script setup>
  const myVariable = 'Bar'
  const dynamicComponent = resolveComponent(myVariable)
</script>
```

**Reply by mnazarani** [reply]
Thank you! The second approach works excellently!  Spent hours figuring it out until found your solution.
